AsyncDisplayKit/Private/_ASPendingState.mm

```diff
@@ -560,95 +558,104 @@ - (void)setAccessibilityIdentifier:(NSString *)newAccessibilityIdentifier
+    layer.anchorPoint = anchorPoint;
 
-  if (_flags.setBounds)
-    layer.bounds = bounds;
+  if (flags.setZPosition)
+    layer.zPosition = zPosition;
 
-  if (_flags.setContentsScale)
+  if (flags.setContentsScale)
     layer.contentsScale = contentsScale;
 
-  if (_flags.setTransform)
+  if (flags.setTransform)
     layer.transform = transform;
 
-  if (_flags.setSublayerTransform)
+  if (flags.setSublayerTransform)
     layer.sublayerTransform = sublayerTransform;
 
-  if (_flags.setContents)
+  if (flags.setContents)
     layer.contents = contents;
 
-  if (_flags.setClipsToBounds)
+  if (flags.setClipsToBounds)
     layer.masksToBounds = clipsToBounds;
 
-  if (_flags.setBackgroundColor)
+  if (flags.setBackgroundColor)
     layer.backgroundColor = backgroundColor;
 
-  if (_flags.setOpaque)
+  if (flags.setOpaque)
     layer.opaque = opaque;
 
-  if (_flags.setHidden)
+  if (flags.setHidden)
     layer.hidden = isHidden;
 
-  if (_flags.setAlpha)
+  if (flags.setAlpha)
     layer.opacity = alpha;
 
-  if (_flags.setCornerRadius)
+  if (flags.setCornerRadius)
     layer.cornerRadius = cornerRadius;
 
-  if (_flags.setContentMode)
+  if (flags.setContentMode)
     layer.contentsGravity = ASDisplayNodeCAContentsGravityFromUIContentMode(contentMode);
 
-  if (_flags.setShadowColor)
+  if (flags.setShadowColor)
     layer.shadowColor = shadowColor;
 
-  if (_flags.setShadowOpacity)
+  if (flags.setShadowOpacity)
     layer.shadowOpacity = shadowOpacity;
 
-  if (_flags.setShadowOffset)
+  if (flags.setShadowOffset)
     layer.shadowOffset = shadowOffset;
 
-  if (_flags.setShadowRadius)
+  if (flags.setShadowRadius)
     layer.shadowRadius = shadowRadius;
 
-  if (_flags.setBorderWidth)
+  if (flags.setBorderWidth)
     layer.borderWidth = borderWidth;
 
-  if (_flags.setBorderColor)
+  if (flags.setBorderColor)
     layer.borderColor = borderColor;
 
-  if (_flags.setNeedsDisplayOnBoundsChange)
+  if (flags.setNeedsDisplayOnBoundsChange)
     layer.needsDisplayOnBoundsChange = needsDisplayOnBoundsChange;
 
-  if (_flags.setAllowsEdgeAntialiasing)
+  if (flags.setAllowsEdgeAntialiasing)
     layer.allowsEdgeAntialiasing = allowsEdgeAntialiasing;
 
-  if (_flags.setEdgeAntialiasingMask)
+  if (flags.setEdgeAntialiasingMask)
     layer.edgeAntialiasingMask = edgeAntialiasingMask;
 
-  if (_flags.needsDisplay)
-    [layer setNeedsDisplay];
-
-  if (_flags.needsLayout)
+  if (flags.needsLayout)
     [layer setNeedsLayout];
 
-  if (_flags.setAsyncTransactionContainer)
+  if (flags.setAsyncTransactionContainer)
     layer.asyncdisplaykit_asyncTransactionContainer = asyncTransactionContainer;
 
-  if (_flags.setOpaque)
+  if (flags.setOpaque)
     ASDisplayNodeAssert(layer.opaque == opaque, @"Didn't set opaque as desired");
-  
-  if (_flags.setFrame)
-    ASDisplayNodeAssert(NO, @"Frame property should only be used for synchronously wrapped nodes.  See setFrame: in ASDisplayNode+UIViewBridge");
+
+  if (flags.setFrame) {
```

Comment:
I'm not totally sure it makes sense to give frame precedence, especially if both bounds and position are set. We may want to trigger an assertion if various overlapping properties are set in the same pending state. This is unfortunate, however, because they are reconciled by CA...

I'm not sure what to recommend here, but it speaks to my concern about the other Area of this patch that modified frame handling. It is by far one of the most tricky parts of the bridge, And is most definitely not as perfect as it could be.